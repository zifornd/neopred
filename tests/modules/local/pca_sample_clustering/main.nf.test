nextflow_process {

    name "Test Process PCA_SAMPLE_CLUSTERING"
    script "modules/local/pca_sample_clustering/main.nf"
    process "PCA_SAMPLE_CLUSTERING"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("$projectDir/assets/samplesheet.csv")
                input[1] = params.batch
                input[2] = params.design
                input[3] = file("$projectDir/tests/test_result/design_batch_tpm.genesymbol.batchremoved.csv")
                input[4] = file("$projectDir/tests/test_result/design_batch_tpm.genesymbol.csv")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out)//.match()
        }

    }

}
