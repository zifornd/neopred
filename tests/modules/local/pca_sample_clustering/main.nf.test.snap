{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "design_batch_pca_plot_after.pdf:md5,f48128bbf1b33ad1f6fd91fc5e55cd74"
                ],
                "1": [
                    "design_batch_pca_plot_before.pdf:md5,82dba73b9e7b9626ec32836415bc24b3"
                ],
                "2": [
                    "versions.yml:md5,90fc01efb982d9f83c172e62d50fed09"
                ],
                "after_pca": [
                    "design_batch_pca_plot_after.pdf:md5,f48128bbf1b33ad1f6fd91fc5e55cd74"
                ],
                "before_pca": [
                    "design_batch_pca_plot_before.pdf:md5,82dba73b9e7b9626ec32836415bc24b3"
                ],
                "versions": [
                    "versions.yml:md5,90fc01efb982d9f83c172e62d50fed09"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-03T05:56:41.124893"
    }
}